### DISCUSSION TITLE
Add Option to scan Qr Code From picture 🖼️ directly on gallery

### DISCUSSION THREAD

**Discussion by sweetoonion** [ente-io/ente#2653] [Enhancements]
Add Option to scan Qr Code From picture 🖼️ directly on gallery like screenshot 
please like Aegis it's more easier ..

![IMG_20240811_021620_171](https://github.com/user-attachments/assets/1a7e49eb-4bb5-47f3-982f-76e4ec053a42)

 most user's don't have second phone to scan QR codes from camera 📷 !! please add it directly so it's easy to import all passwords from other 2fa apps like Google !

**Comment by sweetoonion** [comment]
hope this feature will be added soon 🔜 up vote guys so devs can see the request .. thanks 💟

**Comment by rom4ster** [comment]
I would like to add to this request because I think it is foolhardy to only request something for mobile when you should also have it for desktop. Desktops cannot scan a QR code and most places kind of *dont* want u using desktop for 2fa so they refuse to give out anything other than a QR code.  I think having this feature for Desktop AND Mobile would be awesome.

**Comment by ethsol** [comment]
I was sure reading QR code from file already exists in this fine and comfortable software. 
One more upvote

**Comment by sonpython** [comment]
Everything about the app is great, except for the lack of a feature to scan QR codes from existing photos. Every time I need to scan a code, I have to borrow another phone to take a picture and then scan it, which is inconvenient and poses security risks. I really hope the development team can consider adding this feature soon to make the app more complete.

**Reply by rom4ster** [reply]
Lol everything about this app is great? Have you tried using it on desktop? There are quite a few missing features.

**Reply by sonpython** [reply]
At least it supports next code and export. From authy user :)

**Reply by rom4ster** [reply]
didn't authy have a chrome extension?

**Reply by sonpython** [reply]
It ends supporting chrome extension and desktop app long ago. They also dont allow you to export your key. I have to go to almost 100 websites to recreate the 2fa code

**Reply by rom4ster** [reply]
Fair enough, Ente is really basic on desktop, It cant start as a service and even if it could, it does not allow you to take quick actions or even minimize to tray.

**Comment by GeoQuack** [comment]
I agree doing it from the app would be more straightforward and should be implemented, however, I have found a way to make it work, on an android phone at least:

- Use any QR code reader that can view a picture saved on your device.
- It will generate a URI starting with otpauth://totp/.... 
- Android understands this and ente auth registers itself as a handler for this URI
- Open this URI, the first time you do it you will get a choice to always use ente auth or select from your compatible TOTP apps every time. Choose as appropriate.
- Ente auth will open with your QR code item added to your TOTP codes.
